src/components/chatbox/message.tsx

```diff
@@ -0,0 +1,251 @@
+import { useState, useEffect } from 'react'
+import cn from 'classnames'
+import ReactMarkdown from 'react-markdown'
+import remarkGfm from 'remark-gfm'
+import rehypePrism from '@mapbox/rehype-prism'
+import { MdxPre } from 'components/dev-dot-content/mdx-components/mdx-code-blocks'
+
+// https://helios.hashicorp.design/icons/library
+
+import { IconCheckSquare24 } from '@hashicorp/flight-icons/svg-react/check-square-24'
+import { IconClipboard24 } from '@hashicorp/flight-icons/svg-react/clipboard-24'
+import { IconThumbsDown24 } from '@hashicorp/flight-icons/svg-react/thumbs-down-24'
+import { IconThumbsUp24 } from '@hashicorp/flight-icons/svg-react/thumbs-up-24'
+import { IconWand24 } from '@hashicorp/flight-icons/svg-react/wand-24'
+
+import Button from 'components/button'
+import Text from 'components/text'
+import FeedbackForm from 'components/feedback-form'
+import IconTile from 'components/icon-tile'
+import useAuthentication from 'hooks/use-authentication'
+
+import s from './message.module.css'
+
+export const UserMessage = ({
+	text,
+	image,
+}: {
+	image?: string
+	text: string
+}) => {
+	return (
+		<div className={cn(s.message, s.message_user)}>
+			<div className={cn(s.message_avatar)}>
+				{image ? <img src={image} alt="user avatar" /> : null}
+			</div>
+			<Text /* Body/200/Medium */
+				size={200}
+				weight="medium"
+				className={cn(s.message_content, s.message_user_input)}
+			>
+				{text}
+			</Text>
+			<div className={cn(s.message_gutter)}></div>
+		</div>
+	)
+}
+
+export const AssistantMessage = ({
+	markdown,
+	showActions,
+	conversationId,
+	messageId,
+}: {
+	markdown: string
+	showActions: boolean
+	conversationId: string
+	messageId: string
+}) => {
+	const { session } = useAuthentication()
+	const accessToken = session?.accessToken
+	// Determines green/red button
+	const [rating, setRating] = useState<1 | -1 | 0>(0)
+
+	const handleFeedback = async ({
+		rating,
+		text,
+	}: {
+		rating?: -1 | 1
+		text?: string
+	}) => {
+		const response = await fetch('/api/chat/feedback', {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json',
+				Authorization: `Bearer ${accessToken}`,
+			},
+			body: JSON.stringify({
+				messageId,
+				conversationId,
+				rating,
+				text,
+			}),
+		})
+		return response
```

Comment:
would it be possible to re-update the UI after getting the response? For example, resetting the feedback button state if the request fails?

Reply:
Yes, very possible and very easy — I opted to do nothing (ie. just do an optimistic update and call it a day) because the 👍/👎-request are so non-critical that it's not a big deal which ever way we go.

I'll leave this as-is, but leave the door open to super easy change in the future.